implementations/micrometer-registry-dynatrace/src/test/java/io/micrometer/dynatrace/v2/DynatraceExporterV2Test.java

```diff
@@ -0,0 +1,365 @@
+        assertNotNull(gauge3);
+        List<String> actual3 = exporter.toGauge(gauge3).collect(Collectors.toList());
+        assertThat(actual3).isEmpty();
+    }
+
+    @Test
+    void toGaugeTags() {
+        // invalid name.
+        Gauge.builder("my.gauge", () -> 1.23).tags(Tags.of("tag1", "value1", "tag2", "value2")).register(meterRegistry);
+        Gauge gauge = meterRegistry.find("my.gauge").gauge();
+        assertNotNull(gauge);
+
+        List<String> actual = exporter.toGauge(gauge).collect(Collectors.toList());
+        assertThat(actual).hasSize(1);
+        assertThat(actual.get(0)).contains("tag1=value1").contains("tag2=value2").contains("dt.metrics.source=micrometer");
+        assertThat(actual.get(0)).startsWith("my.gauge,");
+        assertThat(actual.get(0)).hasSize("my.gauge,dt.metrics.source=micrometer,tag1=value1,tag2=value2 gauge,1.23 1617776498381".length());
+    }
+
+    @Test
+    void toCounterInvalidName() {
+        // invalid name.
+        meterRegistry.counter("~~~");
+        Counter counter = meterRegistry.find("~~~").counter();
+        assertNotNull(counter);
+
+        List<String> actual = exporter.toCounter(counter).collect(Collectors.toList());
+        assertThat(actual).isEmpty();
+    }
+
+    @Test
+    void toCounterInvalidCases() {
+        meterRegistry.counter("my.counter1");
+        Counter counter = meterRegistry.find("my.counter1").counter();
+        assertNotNull(counter);
+
+        counter.increment(Double.NaN);
+        List<String> actual1 = exporter.toCounter(counter).collect(Collectors.toList());
+        counter.increment(Double.POSITIVE_INFINITY);
+        List<String> actual2 = exporter.toCounter(counter).collect(Collectors.toList());
+        counter.increment(Double.NEGATIVE_INFINITY);
+        List<String> actual3 = exporter.toCounter(counter).collect(Collectors.toList());
+        assertThat(actual1).hasSize(1);
+        assertThat(actual2).hasSize(1);
+        assertThat(actual3).hasSize(1);
+
+        assertThat(actual1.get(0)).contains("count,delta=0");
+        assertThat(actual2.get(0)).contains("count,delta=0");
+        assertThat(actual3.get(0)).contains("count,delta=0");
+    }
+
+    @Test
+    void toCounterTags() {
+        // invalid name.
+        Counter.builder("my.counter").tags(Tags.of("tag1", "value1", "tag2", "value2")).register(meterRegistry);
+        Counter counter = meterRegistry.find("my.counter").counter();
+        assertNotNull(counter);
+
+        List<String> actual = exporter.toCounter(counter).collect(Collectors.toList());
+        assertThat(actual).hasSize(1);
+        assertThat(actual.get(0)).contains("tag1=value1").contains("tag2=value2").contains("dt.metrics.source=micrometer");
+        assertThat(actual.get(0)).startsWith("my.counter,");
+        assertThat(actual.get(0)).hasSize("my.counter,tag1=value1,dt.metrics.source=micrometer,tag2=value2 count,delta=0 1617796526714".length());
+    }
+
+    @Test
+    void prepareEndpoint() {
```

Comment:
Please also add tests for Managed endpoints (`https://dt.example.com/e/abc12345(/api/v2/metrics/ingest)`).